api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -301,33 +324,32 @@ private boolean checkWorkspaceWriteAccess(String workspaceNamespace, String work
     }
     FirecloudWorkspaceId workspaceId = generateFirecloudWorkspaceId(workspace.getNamespace(),
```

Comment:
You can move this to after the if below. 

Another alternative is to have this method call get() on FC repeatedly until it finds a workspace name that's not taken (rather than trying to create one N times and handling ConflictException.)

I could see going either way.